ext/cedar_policy/src/schema.rs

```diff
@@ -0,0 +1,72 @@
+use std::{collections::HashSet, str::FromStr};
+
+use cedar_policy::{CedarSchemaError, Schema};
+use magnus::{function, method, Error, Module, Object, RArray, RModule, RString, Ruby};
+
+use crate::error::PARSE_ERROR;
+
+#[magnus::wrap(class = "CedarPolicy::Schema")]
+pub struct RSchema(Schema);
+
+impl RSchema {
+    fn new(ruby: &Ruby, schema: RString) -> Result<Self, Error> {
+        if schema.is_empty() {
+            Err(Error::new(
+                ruby.get_inner(&PARSE_ERROR),
+                "you must supply schema contents",
+            ))
+        } else {
+            Self::from_str(&schema.to_string()?)
+                .map_err(|e| Error::new(ruby.get_inner(&PARSE_ERROR), e.to_string()))
+        }
+    }
+
+    pub fn principals(&self) -> RArray {
+        let principals = self.0.principals().collect::<HashSet<_>>();
+        RArray::from_iter(principals.iter().map(|principal| principal.to_string()))
+    }
+
+    pub fn resources(&self) -> RArray {
+        let resources = self.0.resources().collect::<HashSet<_>>();
+        RArray::from_iter(resources.iter().map(|resource| resource.to_string()))
+    }
+
+    pub fn action_groups(&self) -> RArray {
+        RArray::from_iter(self.0.action_groups().map(|group| group.to_string()))
+    }
+
+    pub fn actions(&self) -> RArray {
+        RArray::from_iter(self.0.actions().map(|action| action.to_string()))
+    }
```

Comment:
Any reason to return `string` here? It seems the original method returns an object (e.g. `Entity` or `EntityUid`)

Reply:
We can return  `EntityUid` objects for actions and action groups. I will implement that.

For `principals` and `resources`, the original method returns `EntityTypeName`, which is basically a namespace and a basename. There isn't yet a ruby wrapper for `EntityTypeName`, and a string seemed like a reasonable representation. Would you prefer to have a more faithful and complete representation of the rust API? I can implement that wrapper if you like.